### PR TITLE
BWS-60: + logout

### PR DESCRIPTION
**Pull request by ania-leichenko**

### CODE REVIEW

File: frontend/src/components/app/app.tsx [hunk 1]
```diff
@@ -31,6 +32,7 @@ const App: FC = () => {
           path={AppRoute.ROOT}
           element={<AuthorizedRoute component={<Dashboard />} />}
         />
+        <Route path={AppRoute.LOG_OUT} element={<LogOut />} />
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]:
```suggestion
```
actually you don't need this route at all, you just need to call an action and remove user from the store

File: frontend/src/common/enums/app/app-route.enum.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ enum AppRoute {
   ROOT = '/',
   SIGN_IN = '/sign-in',
   SIGN_UP = '/sign-up',
+  LOG_OUT = '/logout',
```
Comment (what1s1ove) [inline frontend/src/common/enums/app/app-route.enum.ts]:
```suggestion
```

File: frontend/src/components/log-out/log-out.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { FC } from 'react';
```
Comment (what1s1ove) [inline frontend/src/components/log-out/log-out.tsx]: remove file

**Review by what1s1ove — CHANGES_REQUESTED** [review]
`backend/package-lock.json` shouldn't be changed. use the nodejs version from the `readme`

File: frontend/src/components/common/header/header.tsx [hunk 4]
```diff
@@ -6,14 +6,23 @@ import { Link } from 'components/common/common';
...
 const Header: FC = () => {
   const [isVisible, setVisible] = useState<boolean>(false);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
   const { tenant } = useAppSelector(({ app }) => ({
     tenant: app.tenant,
   }));
 
   const handleVisibleChange = (): void => setVisible(!isVisible);
+  const handleVisible = (): void => {
```
Comment (what1s1ove) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
  const handleLogout = (): void => {
```

File: frontend/src/components/common/header/header.tsx [hunk 5]
```diff
@@ -6,14 +6,23 @@ import { Link } from 'components/common/common';
...
+    dispatch(logOut());
+    navigate(AppRoute.SIGN_IN);
```
Comment (what1s1ove) [inline frontend/src/components/common/header/header.tsx]: do it in the action

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/package-lock.json [hunk 6]
```diff
@@ -1,5609 +1,5609 @@
-{
```
Comment (what1s1ove) [inline backend/package-lock.json]: this file sholdn't be changed

File: frontend/src/components/common/header/header.tsx [hunk 7]
```diff
@@ -6,14 +6,22 @@ import { Link } from 'components/common/common';
...
   const [isVisible, setVisible] = useState<boolean>(false);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
   const { tenant } = useAppSelector(({ app }) => ({
     tenant: app.tenant,
   }));
 
   const handleVisibleChange = (): void => setVisible(!isVisible);
+  const handleLogout = (): void => {
+    dispatch(logOut(navigate));
```
Comment (what1s1ove) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
  const dispatch = useDispatch();
  const { tenant } = useAppSelector(({ app }) => ({
    tenant: app.tenant,
  }));

  const handleVisibleChange = (): void => setVisible(!isVisible);
  const handleLogout = (): void => {
    dispatch(logOut());
```

File: frontend/src/store/auth/actions.ts [hunk 8]
```diff
@@ -39,5 +40,13 @@ const loadCurrentUser = createAsyncThunk<
   const { user } = await authApi.getCurrentUser();
   return user;
 });
+const logOut = createAsyncThunk<void, NavigateFunction, AsyncThunkConfig>(
+  ActionType.LOG_OUT,
+  async (navigate, { extra }) => {
+    const { storage } = extra;
+    storage.removeItem(StorageKey.TOKEN);
+    navigate(AppRoute.SIGN_IN);
```
Comment (what1s1ove) [inline frontend/src/store/auth/actions.ts]:
```suggestion
    navigate(AppRoute.SIGN_IN);
```
use navigation service (from `extra`)

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Donymak — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**